packages/x-date-pickers/src/internals/hooks/useDefaultizedTimezone.ts

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import useEventCallback from '@mui/utils/useEventCallback';
+import useControlled from '@mui/utils/useControlled';
+import { useUtils } from './useUtils';
+import type { PickerValueManager } from './usePicker';
+import { PickersTimezone } from '../../models';
+
+const useDefaultizedTimezone = <TDate, TValue>({
+  timezone,
+  value,
+  valueManager,
+}: {
+  timezone: PickersTimezone | undefined;
+  value: TValue;
+  valueManager: PickerValueManager<TValue, TDate, any>;
+}) => {
+  const utils = useUtils<TDate>();
+
+  const timezoneFromValue = React.useMemo(
+    () => valueManager.getTimezone(utils, value),
+    [value, utils, valueManager],
+  );
+
+  const timezoneToRender = timezone ?? timezoneFromValue ?? 'default';
+
+  const setInputTimezone = useEventCallback((newValue: TValue) => {
+    if (timezoneFromValue == null) {
+      return newValue;
+    }
+
+    return valueManager.setTimezone(utils, timezoneFromValue, newValue);
+  });
```

Comment:
Not sure but if `timezoneFromValue` is modified, would `setInputTimezone` be still correct?

Reply:
You mean, if the user changes `props.value` and the new value does not have the same timezone as the previous one ?
Then yes I think it's expected that the from now on `onChange` is fired with the new timezone.

Reply:
It's not a big problem (I'm trying very hard to create bugs) But it leads to inconsistencies when the component is initially `null` and the dev wants to ensure it's only manipulating g UTC dates with 

```jsx
onChange={(newVal) => setValue(dayjs.utc(newVal))}
```

Either you provide the rendering TZ and then the first key pressed is not displayed

https://codesandbox.io/s/nice-kepler-9hnm7l?file=/demo.tsx

https://github.com/mui/mui-x/assets/45398769/0e41f8b2-7dc0-47e9-96b4-ecd142c4fa0c

Or you don't use timezone and then the first key is not pressed, plus when you finish to fill the date, the TZ switch

https://codesandbox.io/s/epic-nova-vfxm5k?file=/demo.tsx

https://github.com/mui/mui-x/assets/45398769/4f800749-d5e6-41fb-b16e-0c877d278f27

Reply:
> Or you don't use timezone and then the first key is not pressed, plus when you finish to fill the date, the TZ switch

https://codesandbox.io/s/sweet-darwin-7tckw4?file=/demo.tsx

Is the following more coherent?